WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -734,6 +741,13 @@ public void onClick(View v) {
                 if (getMySiteFragment() != null) {
                     getMySiteFragment().onActivityResult(requestCode, resultCode, data);
 
+                    boolean isSameSiteSelected = data != null
```

Comment:
Is the `data != null` null check necessary?  It usually makes sense to do before something like `data.getIntExtra`, but removing it [here](https://github.com/wordpress-mobile/WordPress-Android/blob/d60a2e2ac26f570613c91e6f85aaa86c04ad6ab0/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L744) and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/d60a2e2ac26f570613c91e6f85aaa86c04ad6ab0/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L754) doesn't produce lint warnings.

Reply:
My reasoning behind the check is that the `setSite(data)` and `jumpNewPost(data)` that are used right after that do check if the data is null, so I assumed that it actually might be`null`, which could lead to NPE.

Reply:
Removing the null checks from `setSite(data)` and `jumpNewPost(data)` doesn't produce any lint warnings either.  I'm not sure where the first `NullPointerException` would occur if `data` was `null`, but I guess it doesn't hurt to have those null checks.